lib/OpenQA/Git.pm

```diff
@@ -65,22 +65,47 @@ sub _format_git_error ($self, $result, $error_message) {
     return $error_message;
 }
 
+=over 4
+
+=item set_to_latest_master()
+
+This function is used to checkout the latest commit from a git branch provided via update_branch
+and update_remote configuration variables in openqa.ini. In case if both the configuration variables
+are not provided then this function sets the default git branch by first extracting the git URL by using
+get_origin_url and then setting the default git branch from the extracted git URL.
+
+=back
+
+=cut
+
 sub set_to_latest_master ($self, $args = undef) {
     $self->_validate_attributes;
     if (my $update_remote = $self->config->{update_remote}) {
-        my $res = $self->_run_cmd(['remote', 'update', $update_remote]);
-        return $self->_format_git_error($res, 'Unable to fetch from origin master') unless $res->{status};
+        try { $self->fetch($self->get_current_branch) }
```

Comment:
I'm sorry, but in hindsight I think we shouldn't change the behaviour here, because I think it's behaving a little bit different from before.
So my suggestion was wrong.
And I think those settings should be deprecated anyway at some point.
So I would suggest to revert this and the change in the `update_branch` block.